src/hotspot/share/memory/metaspace.hpp

```diff
@@ -234,212 +150,113 @@ class Metaspace : public AllStatic {
     return mdType == ClassType && using_class_space();
   }
 
-  static bool initialized() { return _initialized; }
+  static bool initialized();
 
 };
 
-// Manages the metaspace portion belonging to a class loader
+// A ClassLoaderMetaspace manages MetaspaceArena(s) for a CLD.
+//
+// A CLD owns one MetaspaceArena if UseCompressedClassPointers is false. Otherwise
+// it owns two - one for the Klass* objects from the class space, one for the other
+// types of MetaspaceObjs from the non-class space.
+//
+// +------+       +----------------------+       +-------------------+
+// | CLD  | --->  | ClassLoaderMetaspace | ----> | (non class) Arena |
+// +------+       +----------------------+  |    +-------------------+     allocation top
+//                                          |       |                        v
+//                                          |       + chunk -- chunk ... -- chunk
+//                                          |
+//                                          |    +-------------------+
+//                                          +--> | (class) Arena     |
+//                                               +-------------------+
+//                                                  |
+//                                                  + chunk ... chunk
+//                                                               ^
+//                                                               alloc top
+//
 class ClassLoaderMetaspace : public CHeapObj<mtClass> {
-  friend class CollectedHeap; // For expand_and_allocate()
-  friend class ZCollectedHeap; // For expand_and_allocate()
-  friend class ShenandoahHeap; // For expand_and_allocate()
-  friend class Metaspace;
-  friend class MetaspaceUtils;
-  friend class metaspace::PrintCLDMetaspaceInfoClosure;
-  friend class VM_CollectForMetadataAllocation; // For expand_and_allocate()
-
- private:
 
-  void initialize(Mutex* lock, Metaspace::MetaspaceType type);
-
-  // Initialize the first chunk for a Metaspace.  Used for
-  // special cases such as the boot class loader, reflection
-  // class loader and hidden class loader.
-  void initialize_first_chunk(Metaspace::MetaspaceType type, Metaspace::MetadataType mdtype);
-  metaspace::Metachunk* get_initialization_chunk(Metaspace::MetaspaceType type, Metaspace::MetadataType mdtype);
+  // A reference to an outside lock, held by the CLD.
```

Comment:
Should this be moved to classLoaderMetaspace.hpp (now that you have created a new file classLoaderMetaspace.cpp).